addons/html_editor/static/src/backend/dynamic_field/dynamic_field_plugin.js

```diff
@@ -0,0 +1,371 @@
+import { isHtmlContentSupported } from "@html_editor/core/selection_plugin";
+import { isContentEditable } from "@html_editor/utils/dom_info";
+import { closestElement, visitNode } from "@html_editor/utils/dom_traversal";
+import { leftPos, rightPos } from "@html_editor/utils/position";
+import { withSequence } from "@html_editor/utils/resource";
+import { _t } from "@web/core/l10n/translation";
+
+import { FieldSelectorPopover } from "@html_editor/backend/dynamic_field/field_selector_popover";
+import { QWebPlugin } from "@html_editor/others/qweb_plugin";
+import { Plugin } from "@html_editor/plugin";
+
+const QWEB_T_OUT = ["t-field", "t-out", "t-esc"];
+const DUMMY_CONTENT_ATTRS = ["data-oe-demo", "data-oe-expression-readable"];
+
+const STYLE_QWEB_ATTRS = ["class", "style"]
+    .map((attr) => [`t-att-${attr}`, `t-attf-${attr}`])
+    .flat();
+const STYLE_BUTTON_IDS = [
+    "forecolor",
+    "backcolor",
+    "bold",
+    "italic",
+    "underline",
+    "strikethrough",
+    "remove_format",
+    "font",
+    "font-size",
+    "font-family",
+    "link",
+];
+
+export class DynamicFieldPlugin extends Plugin {
+    static id = "dynamicField";
+    static dependencies = ["selection", "history", "overlay", "dom", "toolbar", QWebPlugin.id];
+
+    /** @type {import("plugins").EditorResources} */
+    resources = {
+        unsplittable_node_predicates: (node) =>
+            node.nodeType === Node.ELEMENT_NODE && node.matches(".page, .header, .footer"),
```

Comment:
Why is the `DynamicFieldPlugin` using a resource to avoid split on header and footer class ?
Does seems strange to me, in term of separation of concern.